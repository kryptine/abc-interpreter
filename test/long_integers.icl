module long_integers

import StdList
import Data.Integer

addIo :: !Int !Int -> (!Bool,!Int);
addIo a b = code inline {
	addIo
}

subIo :: !Int !Int -> (!Bool,!Int);
subIo a b = code inline {
	subIo
}

Start = map toString
	[ a + b
	, a - b
	, a * b
	, a / b
	, a rem b
	, c + d
	, c - d
	, c * d
	, c / d
	, c rem d
	]

a = toInteger "1234567890123456789012345678901234567890"
b = toInteger     "654321098765432109876543210987654321"
c = toInteger "123123"
d = toInteger "456"
