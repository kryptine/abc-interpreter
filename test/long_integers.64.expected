["1235222211222222221122222222112222222211","1233913569024691356902469135690246913569","807803818366101506665142263149367437441208653362292333223746380111126352690","1886","518297851851829785185182978518518484","123579","122667","56144088","270","3"]
halt at 4
563 261581 262144
